src/oracle/oracle.cairo

```diff
@@ -354,82 +370,72 @@ mod Oracle {
         }
     }
 
-
-    impl CheckpointStoreImpl of Store<Checkpoint> {
-        fn read(address_domain: u32, base: StorageBaseAddress) -> SyscallResult<Checkpoint> {
-            let timestamp_base = storage_base_address_from_felt252(
-                storage_address_from_base_and_offset(base, 0_u8).into()
-            );
-            let timestamp: u64 = Store::<u128>::read(address_domain, timestamp_base)?
-                .try_into()
+    impl EntryStorePacking of StorePacking<EntryStorage, felt252> {
+        fn pack(value: EntryStorage) -> felt252 {
+            value.timestamp.into()
+                + value.volume.into() * TIMESTAMP_SHIFT_U32
+                + value.price.into() * VOLUME_SHIFT_U132
+        }
+        fn unpack(value: felt252) -> EntryStorage {
+            let value: u256 = value.into();
+            let volume_shift: NonZero<u256> = integer::u256_try_as_non_zero(
+                VOLUME_SHIFT_U132.into()
+            )
                 .unwrap();
-
-            let value_base = storage_base_address_from_felt252(
-                storage_address_from_base_and_offset(base, 1_u8).into()
-            );
-            let value: u128 = Store::<u128>::read(address_domain, value_base)?;
-            let u8_aggregation_mode: u8 = Store::<felt252>::read(
-                address_domain,
-                storage_base_address_from_felt252(
-                    storage_address_from_base_and_offset(base, 3_u8).into()
-                )
-            )?
-                .try_into()
+            let (price, rest) = integer::u256_safe_div_rem(value, volume_shift);
```

Comment:
Shouldn't we start with the timestamp like `let (timestamp, x) = integer::u256_safe_div_rem(value, timestamp_shift);`

Reply:
We can do so, but is that a meaningful change ? Furthermore, if we want to divide the value by the ```timestamp_shift```, it should rather be:  ```let (rest, timestamp) = integer::u256_safe_div_rem(value, timestamp_shift)``` , no ? The quotient will be the agg of the volume and the price, and the rest the timestamp, wdyt ?

Reply:
According to this, I don't think so ? https://cairopractice.com/posts/store-packing/